doctor-plugin/src/main/java/com/osacky/doctor/JavaHomeCheck.kt

```diff
@@ -3,40 +3,57 @@ package com.osacky.doctor
 import com.osacky.doctor.internal.Finish
 import com.osacky.doctor.internal.PillBoxPrinter
 import org.gradle.api.GradleException
+import org.gradle.api.logging.Logger
 import org.gradle.internal.jvm.Jvm
 import java.io.File
 
 class JavaHomeCheck(
     private val extension: DoctorExtension,
-    private val pillBoxPrinter: PillBoxPrinter
+    private val pillBoxPrinter: PillBoxPrinter,
+    private val logger: Logger
 ) : BuildStartFinishListener {
     override fun onStart() {
-        if (extension.ensureJavaHomeIsSet.get() && environmentJavaHome == null) {
-            throw GradleException(
-                pillBoxPrinter.createPill(
-                    """
-                JAVA_HOME is not set.
-                Please set JAVA_HOME so that switching between Android Studio and the terminal does not trigger a full rebuild.
-                To set JAVA_HOME: (using bash)
-                echo "export JAVA_HOME=${'$'}(/usr/libexec/java_home)" >> ~/.bash_profile
-                or `~/.zshrc` if using zsh.
-                    """.trimIndent()
-                )
-            )
+        val extraMessage = extension.javaHomeHandler.extraMessage.orNull
```

Comment:
```suggestion
        val extraMessage = extension.javaHomeHandler.extraMessage.get()
```
can we just get the value to avoid setting the defautl in two places?